File: spikectl/model/requests/display.py
from spikectl.ujsonrpc import RPCRequest

class ScratchDisplayClearRequest(RPCRequest):

    def __init__(self, idx: str = None):
        super().__init__('scratch.display_clear', {}, idx=idx)


class ScratchDisplaySetPixelRequest(RPCRequest):

    def __init__(self, x: int, y: int, brightness: int, idx: str = None):
        super().__init__('scratch.display_set_pixel', {
            'x': x,
            'y': y,
            'brightness': brightness
        }, idx=idx)


class ScratchDisplayImageRequest(RPCRequest):

    def __init__(self, image: str, idx: str = None):
        super().__init__('scratch.display_image', {
            'image': image
        }, idx=idx)


class ScratchDisplayImageForRequest(RPCRequest):

    def __init__(self, image: str, duration: int, idx: str = None):
        super().__init__('scratch.display_image_for', {
            'image': image,
            'duration': duration
        }, idx=idx)


class ScratchDisplayClearRequest(RPCRequest):

    def __init__(self, idx: str = None):
        super().__init__('scratch.display_clear', {}, idx=idx)


class ScratchCenterButtonLightRequest(RPCRequest):

    def __init__(self, color: int, idx: str = None):
        super().__init__('scratch.center_button_lights', {
            'color': color
        }, idx=idx)

class ScratchDisplayTextRequest(RPCRequest):

    def __init__(self, text: str, idx: str = None):
        super().__init__('scratch.display_text', {
            'text': text
        }, idx=idx)

class ScratchDisplayImageRequest(RPCRequest):

    def __init__(self, image: str = '00000\n00000\n00000\n00000\n00000', idx: str = None):
        super().__init__('scratch.display_image', {
            'image': image
        }, idx=idx)


class ScratchDisplayImageForRequest(RPCRequest):

    def __init__(self, image: str, duration: int, idx: str = None):
        super().__init__('scratch.display_image_for', {
            'image': image,
            'duration': duration
        }, idx=idx)


class ScratchDisplayClearRequest(RPCRequest):

    def __init__(self, idx: str = None):
        super().__init__('scratch.display_clear', {}, idx=idx)


class ScratchDisplaySetPixelRequest(RPCRequest):

    def __init__(self, x: int, y: int, brightness: int, idx: str = None):
        super().__init__('scratch.display_set_pixel', {
            'x': x,
            'y': y,
            'brightness': brightness
        }, idx=idx)

File: spikedev-js/src/requests/display.js
import { RPCRequest } from '../ujsonrpc.js'


export class ScratchDisplaySetPixelRequest extends RPCRequest {

    constructor(x, y, brightness, idx = null) {
        super('scratch.display_set_pixel', {
            'x': x,
            'y': y,
            'brightness': brightness
        }, idx)
    }
}

export class ScratchDisplayClearRequest extends RPCRequest {

    constructor(idx = null) {
        super('scratch.display_clear', {}, idx)
    }
}

export class ScratchCenterButtonLightRequest extends RPCRequest {

    constructor(color, idx = null) {
        super('scratch.center_button_lights', {
            'color': color
        }, idx)
    }
}

export class ScratchDisplayTextRequest extends RPCRequest {

    constructor(text, idx = null) {
        super('scratch.display_text', {
            'text': text
        }, idx)
    }
}

export class ScratchDisplayImageRequest extends RPCRequest {

    constructor(image = '00000\n00000\n00000\n00000\n00000', idx = null) {
        super('scratch.display_image', {
            'image': image
        }, idx)
    }
}

export class ScratchDisplayImageForRequest extends RPCRequest {

    constructor(image, duration, idx = null) {
        super('scratch.display_image_for', {
            'image': image,
            'duration': duration
        }, idx)
    }
}